Content.Server/Objectives/Requirements/IncompatibleConditionsRequirement.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+using Content.Server.Mobs;
+using Content.Server.Objectives.Interfaces;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.Objectives.Requirements
+{
+    public class IncompatibleConditionsRequirement : IObjectiveRequirement
+    {
+        private List<string> _incompatibleConditions = new();
+        public void ExposeData(ObjectSerializer serializer)
+        {
+            serializer.DataField(this, x=>x._incompatibleConditions, "conditions", new List<string>());
+        }
+
+        public bool CanBeAssigned(Mind mind)
+        {
+            foreach (var objective in mind.AllObjectives)
+            {
+                foreach (var condition in objective.Conditions)
+                {
+                    foreach (var incompatibleCondition in _incompatibleConditions)
+                    {
+                        if (incompatibleCondition == condition.GetType().Name) return false;
```

Comment:
This saddens me. :(
Could you make it so conditions expose a string ID like, for example, components do instead?

Reply:
imo this is actually cleaner, since `condition.GetType().Name` is also what i would use to specify the condition in my objectiveprototype yaml